### PR TITLE
fix: deleted tile aware insights

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

If the update payload for an insight includes `dashboards` then we check if there are any newly added or removed dashboards and create or soft-delete tiles to make the dashboard state match the list provided in the payload.

In order to make these changes visible to the activity logger we call `instance.dashboards.set(changes_to_apply, clear=True)`

I added this and it is stupid 😓 

With or without `clear=True` this causes the ORM to delete and recreate tiles as it sees fit. Since we store state on the tiles (e.g. color or layouts) we don't want the ORM to do that 😭 

The effect of this is that making changes to insights can have the unintended side-effect of causing dashboard layouts to change 💣 

## Changes

* no longer does that
* does not allow insight updates that would add an insight to a soft-deleted dashboard

~~NB this breaks the insight activity log. We no longer pick up addition to/removal from activity log.~~

~~But as a first step it is better to have broken activity log than broken dashboard layouts~~

## How did you test this code?

developer test and in the UI

### CODE REVIEW

File: posthog/api/insight.py [hunk 1]
```diff
@@ -423,9 +483,13 @@ def get_queryset(self) -> QuerySet:
                     id__in=DashboardTile.objects.exclude(deleted=True).values_list("dashboard_id", flat=True)
                 ),
             ),
-            "dashboards__created_by",
-            "dashboards__team",
             "dashboards__team__organization",
```
Comment (Twixes) [inline posthog/api/insight.py:486]: Shouldn't this be gone too if we want to stop using raw Django `dashboards`?

File: posthog/api/insight.py [hunk 2]
```diff
@@ -135,13 +143,24 @@ def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Any:
...
         filters = instance.dashboard_filters()
 
         if not filters.get("date_from"):
             filters.update({"date_from": f"-{DEFAULT_DATE_FROM_DAYS}d"})
         representation["filters"] = filters
         return representation
 
+    @lru_cache(maxsize=1)
+    def _dashboard_tiles(self, instance):
+        return [tile.dashboard_id for tile in instance.dashboard_tiles.all()]
```
Comment (Twixes) [inline posthog/api/insight.py:162]: Should we be filtering out the soft-deleted ones here?

**Review by Twixes — COMMENTED** [review]
LGTM though gotta say dashboards are a complex system these days 😅 Got a couple questions

File: posthog/api/insight.py (hunk 1 above)
Comment (pauldambra) [reply posthog/api/insight.py:486]: Ultimately yes, but at the moment it's keeping the number of queries down

File: posthog/api/insight.py (hunk 2 above)
Comment (pauldambra) [reply posthog/api/insight.py:162]: Yep, that adds a query per insight 🤷 . The Django ORM is taunting me at the moment

**Pull request merged**